lib/ecto/query/builder.ex

```diff
@@ -872,20 +873,29 @@ defmodule Ecto.Query.Builder do
   Checks if the field is an atom at compilation time or
   delegate the check to runtime for interpolation.
   """
-  def quoted_field!({:^, _, [expr]}),
-    do: quote(do: Ecto.Query.Builder.field!(unquote(expr)))
-  def quoted_field!(atom) when is_atom(atom),
+  def quoted_atom!(_ast, used_fun \\ "field", used_arity \\ 2)
```

Comment:
Instead of two arguments, have a single argument called used_ref, and you can pass "field/2". Also, remove the default argument and instead make sure that all callers pass the proper reference. :)

Reply:
Addressed in [a85b3fa](https://github.com/elixir-ecto/ecto/pull/3685/commits/a85b3fad4efe11f148988fe8880fdd4da511b3f5).